### PR TITLE
Feature/create new game/151980418

### PR DESCRIPTION
**Pull request by ipheghe**
### What does this PR do?
Users should be able to start a new game session after login or signup 

### Description of Task to be completed?
- It allows user to create a new game session after login or signup. 
- When the user clicks the start game button on the game screen, it alerts the user if he/she wants to start a game with the available number of users
- After the game session has started and a winner is found for the game, log of the current game is created in the database

### How should this be manually tested?
- clone repo or switch to branch 
- run `git pull` on the branch 
- start server on localhost
- navigate to http://localhost:3000/#!/login
- try entering invalid credentials
- click on either play game with strangers or friends button
- it takes the user to the game screen
- click on the start game button and play the game

### Any background context you want to provide?
N/A

## What are the relevant pivotal tracker stories?
story type: feature
story Id: 151980418
story title: Users should be able to create/start a new game

### Screenshots (if appropriate)

image

### Questions:
N/A

### CODE REVIEW

File: app/middlewares/requiresToken.js [hunk 1]
```diff
@@ -2,29 +2,57 @@ const mongoose = require('mongoose'),
...
-const requiresToken = (req, res, next) => {
-  // token could
-  const token = req.headers['x-access-token'] || req.headers.authorization || req.params.token;
-  // the decoded
-  let user;
-  // check if token is passed
-  if (!token) {
-    return res.send(400, { message: 'user authorization token required' });
-  }
+// const requiresToken = (req, res, next) => {
```
Comment (codrex) [inline app/middlewares/requiresToken.js]: Please remove all unwanted code from your work.

Comment (ipheghe) [reply app/middlewares/requiresToken.js]: fixed it

File: public/js/controllers/game.js [hunk 2]
```diff
@@ -174,11 +183,11 @@ angular.module('mean.system')
 
     if ($location.search().game && !(/^\d+$/).test($location.search().game)) {
       console.log('joining custom game');
```
Comment (andela-moe) [inline public/js/controllers/game.js]: I believe this is not necessary for production right? Can we take this `console.log` out of the code

File: public/js/services/game.js [hunk 3]
```diff
@@ -177,6 +177,38 @@ angular.module('mean.system')
...
+    const onError = (err) => {
+      return err;
+    };
+
+    // Notify backend to save game logs When the game has ended
+    socket.on('saveGame', (data) => {
+      if (game.state === 'game ended' && $window.localStorage.token) {
+        const req = {
+          method: 'POST',
+          url: `http://localhost:3000/api/games/${game.gameID}/start`,
```
Comment (andela-moe) [inline public/js/services/game.js]: Why do you have this pointing to your local instance? Please address this

File: public/views/question.html [hunk 4]
```diff
@@ -37,5 +37,43 @@
...
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title" id="exampleModalLongTitle">Cards For Humanity</h5>
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          {{game.players.length}} player(s) cannot play the game
```
Comment (andela-moe) [inline public/views/question.html]: Nice. However, I would advise we use "`{{game.players.length}} player(s) available for this game. You cannot start the game at this time you need a minimum of 3 players to start.`" or something similar

File: public/views/question.html [hunk 5]
```diff
@@ -37,5 +37,43 @@
...
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title" id="exampleModalLongTitle">Cards For Humanity</h5>
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          Are you sure you want to start the game with {{game.players.length}} players?
```
Comment (andela-moe) [inline public/views/question.html]: This can be modified to read "Start the game with {{game.players.length}} players?" or something similar

**Comment by andela-moe** [comment]
Please address comments

File: public/views/question.html (hunk 4 above)
Comment (ipheghe) [reply public/views/question.html]: fixed

File: public/views/question.html (hunk 5 above)
Comment (ipheghe) [reply public/views/question.html]: fixed

File: public/js/services/game.js (hunk 3 above)
Comment (ipheghe) [reply public/js/services/game.js]: fixed

File: public/js/controllers/game.js (hunk 2 above)
Comment (ipheghe) [reply public/js/controllers/game.js]: fixed

File: config/socket/game.js [hunk 6]
```diff
@@ -221,10 +221,19 @@ Game.prototype.stateResults = function(self) {
...
 };
 
+// calls create game log route at the end of a game round
 Game.prototype.stateEndGame = function(winner) {
-  this.state = "game ended";
+  this.state = 'game ended';
   this.gameWinner = winner;
+  const gamePlayers = this.players.map(player => player.username);
   this.sendUpdate();
+  const data = {
```
[the remark this answers is not in the document]
Comment (ipheghe) [reply config/socket/game.js]: fixed

File: app/middlewares/requiresToken.js [hunk 7]
```diff
@@ -1,30 +1,29 @@
...
+  verifyUser(req, res, next) {
+    const token = req.headers['x-access-token'] || req.headers.authorization || req.params.token;
+    if (!token) {
+      return res.status(403).send({
+        message: 'No token provided!'
+      });
+    } else if (token) {
+      jwt.verify(token, process.env.TOKEN_SECRET, (err, decoded) => {
+        if (err) {
+          // return res.status(403).send(err);
```
[the remark this answers is not in the document]
Comment (ipheghe) [reply app/middlewares/requiresToken.js]: fixed

**Pull request closed without merging**